client/session.go

```diff
@@ -1695,9 +1797,10 @@ type blockMetadataReattempt struct {
 }
 
 type blockMetadataReattemptPeerMetadata struct {
-	peer  hostQueue
-	start time.Time
-	size  int64
+	peer     hostQueue
+	start    time.Time
+	size     int64
+	checksum *uint32
```

Comment:
Probably don't need to add checksum to the reattempt peer metadata.  Checksums aren't used for selecting the best block on reattempting fetching a block.